File: src/main/scala/org/nygenome/als/graphdb/value/NeurobankSubjectTimepointProperty.scala
package org.nygenome.als.graphdb.value

import org.apache.commons.csv.CSVRecord

case class NeurobankSubjectTimepointProperty(
                                    timepointEventId:Int,
                                    propertyCategory:String,
                                    propertyCode:String,
                                    propertyName:String,
                                    propertyValue:String,
                                    timepointId:Int
                                       ) {
  val id:String = propertyCategory + ":" +propertyCode

}
object NeurobankSubjectTimepointProperty extends ValueTrait {
  def parseCSVRecord(record:CSVRecord):NeurobankSubjectTimepointProperty = {
    new NeurobankSubjectTimepointProperty(
      record.get("timepoint_event_id").toInt,
      record.get("property_category"),
      record.get("property_code"),
      record.get("property_name"),
      record.get("property_value"),
      record.get("timepoint_id").toInt
    )
  }
}

File: src/main/kotlin/org/biodatagraphdb/alsdb/model/NeurobankSubjectTimepointProperty.kt
package org.biodatagraphdb.alsdb.model

import org.apache.commons.csv.CSVRecord

data class NeurobankSubjectTimepointProperty(
        val timepointEventId: Int,
        val propertyCategory: String,
        val propertyCode: String,
        val propertyName: String,
        val propertyValue: String,
        val timepointId: Int
) {
    companion object : AlsdbModel {
        fun parseCSVRecord(record: CSVRecord): NeurobankSubjectTimepointProperty =
                NeurobankSubjectTimepointProperty(
                        parseValidIntegerFromString(record.get("timepoint_event_id")),
                        record.get("property_category"),
                        record.get("property_code"),
                        record.get("property_name"),
                        record.get("property_value"),
                        parseValidIntegerFromString(record.get("timepoint_id"))
                )
    }
}